include/boost/basic_any.hpp

```diff
@@ -0,0 +1,440 @@
+        static void create(basic_any& any, ValueType&& value, boost::false_type)
+        {
+            any.man = &large_manager<DecayedType>;
+            any.content.large_value = new DecayedType(static_cast<ValueType&&>(value));
+        }
+#endif
+        public: // non-type template parameters accessors
+            static BOOST_CONSTEXPR_OR_CONST std::size_t buffer_size = OptimizeForSize;
+            static BOOST_CONSTEXPR_OR_CONST std::size_t buffer_align = OptimizeForAlignment;
+
+        public: // structors
+
+        basic_any() BOOST_NOEXCEPT
+            : man(0), content()
+        {
+        }
+template <typename T>
+struct print;
+        template<typename ValueType>
+        basic_any(const ValueType & value)
+            : man(0), content()
+        {
+            create(*this, value, is_small_object<ValueType>());
+        }
+
+        basic_any(const basic_any & other)
+          : man(0), content()
+        {
+            if (other.man)
+            {
+                other.man(Copy, *this, &other, 0);
+            }
+        }
+
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+        // Move constructor
+        basic_any(basic_any&& other) BOOST_NOEXCEPT
+          : man(other.man), content()
+        {
+            man(Move, *this, &other, 0);
+            other.man = 0;
+        }
+
+        // Perfect forwarding of ValueType
+        template<typename ValueType>
+        basic_any(ValueType&& value
+            , typename boost::disable_if<boost::is_same<basic_any&, ValueType> >::type* = 0 // disable if value has type `basic_any&`
+            , typename boost::disable_if<boost::is_const<ValueType> >::type* = 0) // disable if value has type `const ValueType&&`
+          : man(0), content()
+        {
+            content.large_value = 0;
+            create(*this, static_cast<ValueType&&>(value), is_small_object<BOOST_DEDUCED_TYPENAME boost::decay<ValueType>::type>());
+        }
+#endif
+
+        ~basic_any() BOOST_NOEXCEPT
+        {
+            if (man)
+            {
+                man(Destroy, *this, 0, 0);
+            }
+        }
+
+    public: // modifiers
+
+        basic_any & swap(basic_any & rhs) BOOST_NOEXCEPT
+        {
+            if (this != &rhs)
+            {
+                if (man && rhs.man)
+                {
+                    basic_any tmp;
+                    rhs.man(Move, tmp, &rhs, 0);
+                    man(Move, rhs, this, 0);
+                    tmp.man(Move, *this, &tmp, 0);
+                }
+                else if (man)
+                {
+                    man(Move, rhs, this, 0);
+                }
+                else if (rhs.man)
+                {
+                    rhs.man(Move, *this, &rhs, 0);
+                }
+            }
+            return *this;
+        }
+
+
+#ifdef BOOST_NO_CXX11_RVALUE_REFERENCES
+        template<typename ValueType>
```

Comment:
Kill that whole branch and just assume that we have rvalues.

Reply:
As we discussed for potential boost::any replacement let's leave composability with C++98